pkg/api/v6/api.go

```diff
@@ -58,6 +58,7 @@ type GitConfig struct {
 	Remote       GitRemoteConfig   `json:"remote"`
 	PublicSSHKey ssh.PublicKey     `json:"publicSSHKey"`
 	Status       git.GitRepoStatus `json:"status"`
+	Error        string            `json:"errors"`
```

Comment:
This will break backwards compatibility with clients using the v6 endpoint.

You would need to jump the API version to extend it. But, even then, I am not sure that's the best place for the error (to be fair I don't think that the status should be part of the config either, bit that's how it is).

Reply:
Ah right, thank you for reviewing.
I can think of making `git.GitRepoStatus` a map containing the `Error`, or adding another endpoint for the repo status?

Reply:
After taking a look at it, adding the error is probably OK but you should add a new version of the data structure and a new version of the endpoint (which will, in turn, require bumping the latest version of the API)

Reply:
Ok, I will update the PR

Reply:
Since the Error field is optional, I don't think just adding it to the v6 API response breaks backward-compatibility. A way to reason about it is to do the matrix of old version / new version and see if mismatches will cause a problem.

In the following table, "Old" means "does not know about the Error field" and "New" means "expects to use the Error field".

|                    | Old fluxd | New fluxd |
|-----------------|------|--------|
| **Old fluxctl**| Operates as now | Will not deserialise Error field, and thereby operate as now |
| **New fluxctl** | Can act conditionally on presence of Error field | Operates as desired |

Reply:
Well, I was thinking about other clients other than `fluxctl`. I know there is people using the API, and their JSON parsers may be stricts or not do well with new fields.

If we don't care about those usecases then I guess it would be fine.

Reply:
Hmm yes true, my thinking is limited to the era before people started using the API directly. OK, new API version it is, ulp.

Reply:
On third thought -- it is a fair amount of trouble to percolate a new version through intermediating code (e.g., Weave Cloud, but no doubt other things). I would be prepared to accept the potential outcry over breaking strict parsing, as a trade for a smaller change overall. wdyt, when it's put in those terms, @2opremio ?

Reply:
OK, I think that's fine :) , @dinosk is (rightfully) going to hate me for it though.

Reply:
Sorry @dinosk! Blame me, I'm the trouble-maker.

So: to be clear, let's go back to how you initially had it, with an Error field added to `v6.GitConfig`, and that field being used if present in `cmd/fluxctl/sync_cmd.go`.

Reply:
😅 it is no problem! I see your point, I’ll revert it to the initial changes

Reply:
_makes backup of new api version_
Ok, this is back to the initial changes

Reply:
Thanks for being so understanding!